### ISSUE TITLE
Deploying --service-discovery does not see kubefedctl in current dir

### ISSUE THREAD

**Issue by manosnoam** [submariner-io/submariner-operator#166]
When installing Service Discovery (lighthouse and kubefed-operator), it will fail if kubfedctl binary is not found in $PATH, even when it is found on current directory:

```
$ cd submariner-operator
$ ls -l kubefedctl
    -rwxr-xr-x 1 nmanos nmanos 42319878 Aug 17 06:55 kubefedctl

$ ./bin/subctl deploy-broker --service-discovery --clusterid subm-cluster-a

 • Deploying broker  ...
 ✓ Deploying broker
 • Creating broker-info.subm file  ...
 ✓ Creating broker-info.subm file
 ✓ A new IPSEC PSK will be generated for broker-info.subm
 • Deploying Service Discovery controller  ...
 ✗ Deploying Service Discovery controller
 ✓ Created operator CRDs
 ✓ Created operator namespace: kubefed-operator
 ✓ Created operator service account
 ✓ Deployed the operator successfully

Failed to deploy Service Discovery controller: error deploying KubeFed: error enabling namespaces in federation: exec: "kubefedctl": executable file not found in $PATH

subctl version: dev
built from git commit id: 77d0a1c3dcbf0d6520706937191b4eb729a71eb3
```

Please also search for kubefedctl in current directory, so user can download it locally, without installing to system path (e.g. /usr/local/bin/ )which requires sudo. 

Note: This issue is relevant as long as subctl relies on kubefedctl binary.

**Comment by skitt** [comment]
> Please also search for kubefedctl in current directory, so user can download it locally, without installing to system path (e.g. /usr/local/bin/ )which requires sudo.

Well, the user can also add the appropriate directory to their PATH, *e.g.* `~/go/bin` or `~/.local/bin`, they don’t need `sudo`...

(I agree we need to deal with this better.)

**Comment by manosnoam** [comment]
This issue is not resolved with #169 (b8b507f).
Even after downloading and installing kubefedctl:
```
$ cp kubefedctl ~/go/bin/
$ kubefedctl version
kubefedctl version: version.Info{Version:"v0.1.0-rc6", GitCommit:"7586b42f4f477f1912caf28287fa2e0a7f68f407", GitTreeState:"clean", BuildDate:"2019-08-17T03:55:05Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

When building subctl, it searches for the GO package of Kubefed, but since I only downloaded and installed the kubefedctl binary to PATH, **without** running go get/install of kubfed repository, then "github.com/kubernetes-sigs/kubefed" does not exists, and build fails:

```

[submariner-operator]$ make bin/subctl
...
[submariner-operator]$ CGO_ENABLED=0 go build -ldflags -X github.com/submariner-io/submariner-operator/pkg/version.Version=dev  -X github.com/submariner-io/submariner-operator/pkg/version.Commit=b8b507ff4e70d6974c2965f9a166162ef6d5ffba -o bin/subctl-dev-linux-amd64 ./pkg/subctl/main.go
build command-line-arguments: cannot load github.com/kubernetes-sigs/kubefed/pkg/version: open /go/src/github.com/submariner-io/submariner-operator/vendor/github.com/kubernetes-sigs/kubefed/pkg/version: no such file or directory
[submariner-operator]$ ./scripts/build-subctl onetime false
time="2020-02-16T13:51:55+02:00" level=fatal msg="exit status 1"
make[1]: *** [build-subctl] Error 1
make[1]: Leaving directory `/home/nmanos/go/src/github.com/submariner-io/submariner-operator'
make: *** [bin/subctl] Error 2



```

**Comment by manosnoam** [comment]
Seems like ./pkg/subctl/lighthouse/deploy/ensure.go is searching for:
"github.com/submariner-io/submariner-operator/pkg/subctl/operator/kubefed"

but it was not yet created.

**Comment by skitt** [comment]
The vendoring issue should be fixed by #174.

**Comment by mangelajo** [comment]
Yes that error was by the lack of a "make vendor" being ran.

#174 should do that automatically for you even if you forget.
